app/gui/src/main/scala/org/bitcoins/gui/WalletGUIModel.scala

```diff
@@ -0,0 +1,79 @@
+package org.bitcoins.gui
+
+import org.bitcoins.cli.CliCommand.{GetBalance, GetNewAddress, SendToAddress}
+import org.bitcoins.cli.ConsoleCli
+import org.bitcoins.core.currency.Bitcoins
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.gui.dialog.{GetNewAddressDialog, SendDialog}
+import scalafx.beans.property.{ObjectProperty, StringProperty}
+import scalafx.scene.control.Alert.AlertType
+import scalafx.scene.control.Alert
+import scalafx.stage.Window
+
+import scala.util.{Failure, Success}
+
+class WalletGUIModel() {
+  var taskRunner: TaskRunner = _
+
+  // Sadly, it is a Java "pattern" to pass null into
+  // constructors to signal that you want some default
+  val parentWindow: ObjectProperty[Window] =
+    ObjectProperty[Window](null.asInstanceOf[Window])
+
+  updateBalance()
+
+  def onGetNewAddress(): Unit = {
+    val address = StringProperty("")
+
+    taskRunner.run(
+      caption = "Get New Address",
+      op = {
+        ConsoleCli.exec(GetNewAddress) match {
+          case Success(commandReturn) => address.value = commandReturn
+          case Failure(err)           => throw err
+        }
+      }
+    )
+
+    GetNewAddressDialog.showAndWait(parentWindow.value, address)
+  }
+
+  def onSend(): Unit = {
+    val result = SendDialog.showAndWait(parentWindow.value)
+
+    result match {
+      case Some((address, amount)) =>
+        taskRunner.run(
+          caption = s"Send $amount to $address",
+          op = {
+            ConsoleCli.exec(
+              SendToAddress(BitcoinAddress(address).get,
+                            Bitcoins(BigDecimal(amount)))) match {
+              case Success(txid) =>
+                GlobalData.log.value =
+                  s"Sent $amount to $address in tx: $txid\n\n${GlobalData.log()}"
+              case Failure(err) => throw err
+            }
+          }
+        )
+      case None => ()
+    }
+
+    updateBalance()
+  }
+
+  private def updateBalance(): Unit = {
+    ConsoleCli.exec(GetBalance(isSats = false)) match {
```

Comment:
Controversial!! No sats the standard? :P

Reply:
Petition for eventual config flag

Reply:
I figured I would just mimic the CLI for now (and this is its default value), in the future I imagine we can add a checkbox or dropdown or settings of some kind to allow this to be specified